scipy/optimize/tests/test_zeros.py

```diff
@@ -302,6 +306,19 @@ def test_root_scalar_fail(self):
         with pytest.raises(ValueError):
             root_scalar(f1, method='halley', fprime2=f1_2, x0=3, xtol=1e-6)  # no fprime
 
+    def test_array_chandrupatla(self):
+        """test chandrupatla with array, similar to newton method"""
+
+        def f1(x, a):
+            return a-x**2
+
+        k = np.arange(1,8)
+        result = zeros.chandrupatla(f1, 0, 3, args=(k,))
+        x_expected = ([1., 1.41421356, 1.73205081,
+                       2., 2.23606798, 2.44948974,
+                       2.64575131])
+        assert_allclose(result.x, x_expected)
```

Comment:
Since this test mentions that `newton` is a similar root finder, I tried to play around locally with a similar call to `newton` in this test:

```diff
--- a/scipy/optimize/tests/test_zeros.py
+++ b/scipy/optimize/tests/test_zeros.py
@@ -317,6 +317,7 @@ class TestBasic:
         x_expected = ([1., 1.41421356, 1.73205081,
                        2., 2.23606798, 2.44948974,
                        2.64575131])
+        result2 = zeros.newton(f1, x0=0, x1=3, args=(k,))
         assert_allclose(result.x, x_expected)
```

which complains about `E   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`.

I'm guessing that's a feature if `chandrupatla` is meant to handle vector arguments better or something like that.

It got me thinking though, if it isn't already covered by the testing harnesses Matt mentioned, that spot checking equivalence to `newton` in a few cases might not hurt, for example another test that looks like this (this does pass) parametrized over a few values of `k` perhaps:

```diff
         k = np.arange(1,8)
-        result = zeros.chandrupatla(f1, 0, 3, args=(k,))
-        x_expected = ([1., 1.41421356, 1.73205081,
-                       2., 2.23606798, 2.44948974,
-                       2.64575131])
-        assert_allclose(result.x, x_expected)
+        result = zeros.chandrupatla(f1, 0, 3, args=(k[0],))
+        expected = zeros.newton(f1, x0=0, x1=3, args=(k[0],))
+        assert_allclose(result.x, expected)
+

```

Reply:
Thanks so much for the review! I'll take a look at what you mentioned and add that in.